build/Microsoft.DotNet.Cli.Compile.targets

```diff
@@ -0,0 +1,180 @@
+
+  <UsingTask TaskName="FixModeFlags" AssemblyFile="$(CLIBuildDll)" />
+  <UsingTask TaskName="ChangeEntryPointLibraryName" AssemblyFile="$(CLIBuildDll)" />
+  <UsingTask TaskName="RemoveAssetFromDepsPackages" AssemblyFile="$(CLIBuildDll)" />
+  <UsingTask TaskName="CleanPublishOutput" AssemblyFile="$(CLIBuildDll)" />
+  <UsingTask TaskName="SharedFrameworkNameVersionPath" AssemblyFile="$(CLIBuildDll)" />
+  <UsingTask TaskName="CrossgenDirectory" AssemblyFile="$(CLIBuildDll)" />    
+
+  <Target Name="Compile" DependsOnTargets="BuildDotnetCliBuildFramework;
+                                           Init;
+                                           SetupStage;
+                                           CompileStage;
+                                           BuildProjectsForNuGetPackages;
+                                           GenerateNuGetPackagesArchive">
+  </Target>
+
+  <Target Name="SetupStage">
+    <ItemGroup>
+      <CompileStageInputs Include="$(RepoRoot)/src/**/*.cs" Exclude="$(RepoRoot)/src/**/obj/**/*.cs" />
+      <CompileStageInputs Include="$(RepoRoot)/src/**/project.json;$(RepoRoot)/src/**/project.lock.json" />
+
+      <Stage Include="Stage1">
+        <Inputs>@(CompileStageInputs)</Inputs>
+        <StageDirectory>$(Stage1Directory)</StageDirectory>
+        <StageSymbolsDirectory>$(Stage1SymbolsDirectory)</StageSymbolsDirectory>
+        <DotnetExe>$(Stage0Path)/dotnet.exe</DotnetExe>
+      </Stage>
+      <Stage Include="Stage2">
+        <Inputs>@(CompileStageInputs)</Inputs>
+        <StageDirectory>$(Stage2Directory)</StageDirectory>
+        <StageSymbolsDirectory>$(Stage2SymbolsDirectory)</StageSymbolsDirectory>
+        <DotnetExe>$(DotnetStage1)</DotnetExe>
+      </Stage>          
+    </ItemGroup>        
+  </Target>    
+
+  <Target Name="CompileStage"
+          Inputs="%(Stage.Identity);%(Stage.Inputs)"
+          Outputs="%(StageDirectory)/sdk/$(NugetVersion)/dotnet.dll">
+      <PropertyGroup>
+        <SrcDirectory>$(RepoRoot)/src</SrcDirectory>
+
+        <ExeSuffix Condition=" '$(OSName)' == 'win' ">.exe</ExeSuffix>
+        <ExeSuffix Condition=" '$(OSName)' != 'win' "></ExeSuffix>
+
+        <DynamicLibPrefix Condition=" '$(OSName)' == 'win' "></DynamicLibPrefix>
+        <DynamicLibPrefix Condition=" '$(OSName)' != 'win' ">lib</DynamicLibPrefix>
+
+        <DynamicLibSuffix Condition=" '$(OSName)' == 'win' ">.dll</DynamicLibSuffix>
+        <DynamicLibSuffix Condition=" '$(OSName)' == 'osx' ">.dylib</DynamicLibSuffix>
+        <DynamicLibSuffix Condition=" '$(DynamicLibSuffix)' == '' ">.so</DynamicLibSuffix>
+
+        <DotnetHostBaseName>dotnet$(ExeSuffix)</DotnetHostBaseName>
+        <DotnetHostFxrBaseName>$(DynamicLibPrefix)hostfxr$(DynamicLibSuffix)</DotnetHostFxrBaseName>
+        <HostPolicyBaseName>$(DynamicLibPrefix)hostpolicy$(DynamicLibSuffix)</HostPolicyBaseName>
+
+        <StageDirectory>%(Stage.StageDirectory)</StageDirectory>
```

Comment:
This is used in some transformations below (Copy) that will have a different itemgroup on it. If we try to access this with the % approach, it will try to use it from the files and not the batch. That's why I set it as a property.